.github/workflows/pull-request.yml

```diff
@@ -3,26 +3,56 @@ name: Pull Request Worfklow
 on: [pull_request]
 
 jobs:
-  build-signing-gateway:
```

Comment:
@wal33d006 here are some important changes: I've made github actions run checks in folders only if something has changed in them compared to the base branch.